### PR TITLE
API v2: Deprecate steps and replace with values on runs endpoint

### PR DESCRIPTION
**Pull request by rowanseymour**
If we're going the direction of compressing steps but leaving values intact - makes sense to have the API expose values rather than steps. This PR provides a very simple view of run values - if we can ensure that UReport etc can work with this then we should be good.

### CODE REVIEW

File: temba/api/tests/test_v2.py [hunk 1]
```diff
@@ -1544,7 +1544,7 @@ def test_runs(self):
         frank_run2.refresh_from_db()
 
         # no filtering
-        with self.assertNumQueries(NUM_BASE_REQUEST_QUERIES + 7):
```
Comment (rowanseymour) [inline temba/api/tests/test_v2.py]: Two extra prefetches to get values and rulesets

File: temba/flows/models.py [hunk 2]
```diff
@@ -3090,6 +3090,10 @@ class RuleSet(models.Model):
     def get(cls, flow, uuid):
         return RuleSet.objects.filter(flow=flow, uuid=uuid).select_related('flow', 'flow__org').first()
 
+    @property
+    def context_key(self):
```
Comment (rowanseymour) [inline temba/flows/models.py]: First step toward making this more of thing in its own right (not 100% sure about name)

Comment (norkans7) [inline temba/flows/models.py]: ruleset-label-slug

Comment (rowanseymour) [inline temba/flows/models.py]: I think moving forward, the fact that it is a slugged version of the ruleset label, is something specific to RapidPro - the flow spec should allow people to explicitly define the variable in the context to be updated - and eventually updating the context becomes an explicit action in itself.

Comment (ericnewcomer) [inline temba/flows/models.py]: Correct, we don't want to make any connection to the notion of rulesets with values.

**Comment by ericnewcomer** [comment]
This is much much closer. I do wonder a bit about the future scenario when flow results are not lockstep with branching. In this case, there's no category. It's just what was explicitly saved. So take for example a question asking age.

Today, we might give:
`age: { value: 17, category: "Youth" }`

These are both useful things, however, in nextgen flows it might look more like:

Save "age" as @flow.age
Save "age_group" as @flow.age.category

Yielding instead:
`values: { age: 17, age_group: 'Youth' }`

Or just one or the other depending on what was saved. Considering RuleSets are effectively implicit value saving, it's not wholly unreasonable for those to be flattened and treated as two values saved implicitly.

So something like this might be more future-proof, if slightly awkward now.
`{ age: 17, age_category: "Youth" }`

Thoughts?

**Comment by ericnewcomer** [comment]
This is much much closer. I do wonder a bit about the future scenario when flow results are not lockstep with branching. In this case, there's no category. It's just what was explicitly saved. So take for example a question asking age.

Today, we might give:
`age: { value: 17, category: "Youth" }`

These are both useful things, however, in nextgen flows it might look more like:

Save "age" as @flow.age
Save "age_group" as @flow.age.category

Yielding instead:
`values: { age: 17, age_group: 'Youth' }`

Or just one or the other depending on what was saved. Considering RuleSets are effectively implicit value saving, it's not wholly unreasonable for those to be flattened and treated as two values saved implicitly.

So something like this might be more future-proof, if slightly awkward now.
`{ age: 17, age_category: "Youth" }`

**Comment by rowanseymour** [comment]
Hmm yeah certainly some good reasons for wanting flow output to be flat - simpler from an analytics point of view.

Am a little concerned about colliding with existing value names if we just suffix "_category" onto value names. Checked and both prod instances have some rulesets (~7 each) with labels ending this way already.

**Pull request merged**